instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -651,6 +666,12 @@ def __init__(
             )
             or []
         )
+        self.exclude_receive_span = exclude_receive_span or strtobool(
+            os.getenv(OTEL_PYTHON_ASGI_EXCLUDE_RECEIVE_SPAN, "false")
```

Comment:
Is it possible for multiple instrumentations that depend on `asgi` middleware to be used (like [starlette](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py#L226) and [fastapi](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py#L263)? What if a user wants different internal http span configurations per instrumentation? Should we care about this scenario or does this seem not likely?

Reply:
That's fair, the env vars would certainly be global. If you wanted different behavior for different instrumentations then you'd need to configure them individually.

Reply:
As you've mentioned in the above [conversation](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2802#discussion_r1722003261), adding this as an explicit configuration setting in each instrumentation might be the way to go for now instead of `asgi` middleware. Feel free to change your PR just for `fastapi` :)